File: backend/internal/router/rbac.go
package router

import (
	"fayhub/internal/controller"
	"fayhub/internal/middleware"

	"github.com/gin-gonic/gin"
)

// RBACRouter RBAC权限路由
// @Summary RBAC权限路由
// @Description 注册角色、权限相关的路由
// @Tags 路由管理
type RBACRouter struct{}

// Init 初始化RBAC路由
// @Summary 初始化RBAC路由
// @Description 注册角色管理、权限分配等接口
// @Tags 路由管理
func (s *RBACRouter) Init(router *gin.Engine) {
	rbacGroup := router.Group("/api/rbac")
	rbacGroup.Use(middleware.JwtAuthMiddleware())
	rbacGroup.Use(middleware.TenantMiddleware())

	rbacGroup.POST("/roles", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.RBACController.CreateRole)
	rbacGroup.GET("/roles", controller.ControllerGroupApp.RBACController.GetRoleList)
	rbacGroup.GET("/roles/:roleID", controller.ControllerGroupApp.RBACController.GetRoleByID)
	rbacGroup.PUT("/roles/:roleID", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.RBACController.UpdateRole)
	rbacGroup.DELETE("/roles/:roleID", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.RBACController.DeleteRole)

	rbacGroup.POST("/assign-role", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.RBACController.AssignRoleToUser)
	rbacGroup.POST("/remove-role", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.RBACController.RemoveRoleFromUser)
	rbacGroup.GET("/users/:userID/roles", controller.ControllerGroupApp.RBACController.GetUserRoles)
	rbacGroup.GET("/users/:userID/permissions", controller.ControllerGroupApp.RBACController.GetUserPermissions)
}

File: web/src/api/rbac.ts
import request, { type ApiResponse, type PageResult, type PageParams } from './request'

export interface Role {
  id: number
  name: string
  type: number
  description: string
  status: number
  data_scope: number
  tenant_id: number
  created_at: string
  updated_at: string
}

export interface CreateRoleParams {
  name: string
  type: number
  description?: string
  data_scope?: number
}

export interface UpdateRoleParams {
  name?: string
  type?: number
  description?: string
  data_scope?: number
}

export interface RoleListParams extends PageParams {}

export interface AssignRoleParams {
  user_id: number
  role_id: number
}

export interface RemoveRoleParams {
  user_id: number
  role_id: number
}

export interface Permission {
  menus: string[]
  apis: string[]
}

const rbacApi = {
  createRole(params: CreateRoleParams): Promise<ApiResponse<Role>> {
    return request.post('/rbac/roles', params)
  },

  getRoleList(params?: RoleListParams): Promise<ApiResponse<PageResult<Role>>> {
    return request.get('/rbac/roles', { params })
  },

  getRoleByID(roleID: number): Promise<ApiResponse<Role>> {
    return request.get(`/rbac/roles/${roleID}`)
  },

  updateRole(roleID: number, params: UpdateRoleParams): Promise<ApiResponse<Role>> {
    return request.put(`/rbac/roles/${roleID}`, params)
  },

  deleteRole(roleID: number): Promise<ApiResponse<null>> {
    return request.delete(`/rbac/roles/${roleID}`)
  },

  assignRoleToUser(params: AssignRoleParams): Promise<ApiResponse<null>> {
    return request.post('/rbac/assign-role', params)
  },

  removeRoleFromUser(params: RemoveRoleParams): Promise<ApiResponse<null>> {
    return request.post('/rbac/remove-role', params)
  },

  getUserRoles(userID: number): Promise<ApiResponse<Role[]>> {
    return request.get(`/rbac/users/${userID}/roles`)
  },

  getUserPermissions(userID: number): Promise<ApiResponse<Permission>> {
    return request.get(`/rbac/users/${userID}/permissions`)
  }
}

export default rbacApi
